### ISSUE TITLE
Hot Exit Tests don't return useful errors

### ISSUE THREAD

**Issue by Lixire** [Microsoft/vscode#29064]
- In vscode/src/vs/workbench/services/textfile/test/textFileService.test.ts
- If you fail an assertion then you don't get useful feedback
- To fail tests, can call `this.contextService.getWorkspace().resource.toString()` in loadFromFile function in `textFileEditorModel.ts`

![image](https://user-images.githubusercontent.com/13241824/27311616-f889db84-5516-11e7-8d57-c6a5e308551b.png)

**Comment by Tyriar** [comment]
Yeah you'll need to debug into them if this happens. A timeout occurs which means that the expected conditions never occurs, not much more we could report.

**Comment by roblourens** [comment]
We were debugging this and the problem is simply that an assert is failing, but the exception is lost in a promise chain. If you handle it correctly you can get an actual error message.

**Comment by Tyriar** [comment]
PR welcome :smile:
